consensus/src/lib.rs

```diff
@@ -485,36 +493,41 @@ pub struct Proposer<C: Send + Sync, TxApi: PoolChainApi> where
 	transaction_pool: Arc<Pool<TxApi>>,
 }
 
-impl<C, TxApi> consensus::Proposer<Block, AuraConsensusData> for Proposer<C, TxApi> where
+impl<C, TxApi> consensus::Proposer<Block> for Proposer<C, TxApi> where
 	TxApi: PoolChainApi<Block=Block>,
 	C: ProvideRuntimeApi + HeaderBackend<Block> + Send + Sync,
-	C::Api: ParachainHost<Block> + BlockBuilderApi<Block, InherentData>,
+	C::Api: ParachainHost<Block> + BlockBuilderApi<Block>,
 {
 	type Error = Error;
-	type Create = Either<
-		CreateProposal<C, TxApi>,
-		future::FutureResult<Block, Error>,
-	>;
+	type Create = Either<CreateProposal<C, TxApi>, future::FutureResult<Block, Error>>;
 
-	fn propose(&self, consensus_data: AuraConsensusData) -> Self::Create {
+	fn propose(&self, inherent_data: InherentData) -> Self::Create {
 		const ATTEMPT_PROPOSE_EVERY: Duration = Duration::from_millis(100);
 		const SLOT_DURATION_DENOMINATOR: u64 = 3; // wait up to 1/3 of the slot for candidates.
 
 		let initial_included = self.tracker.table.includable_count();
 		let now = Instant::now();
 
+		let aura_slot = match inherent_data.aura_inherent_data() {
+			Ok(slot) => slot,
+			Err(e) => return Either::B(future::err(ErrorKind::InherentError(e).into())),
+		};
+
 		let dynamic_inclusion = DynamicInclusion::new(
 			self.tracker.table.num_parachains(),
 			self.tracker.started,
-			Duration::from_secs(consensus_data.slot_duration / SLOT_DURATION_DENOMINATOR),
+			Duration::from_secs(aura_slot / SLOT_DURATION_DENOMINATOR),
```

Comment:
that's incorrect; the slot duration is way different than the slot number. We probably don't need a strict dependence on the `runtime_aura` stuff in fact, only to change `Proposer` to be parameterized by `slot_duration` which is determined at startup.

Reply:
(what we lose by that parameterization is the ability to adapt to changing blocktimes, but it is not urgent to do that for now)